### PR TITLE
[Unity] Add bind_constants option to FuseOpsByPattern

### PR DESCRIPTION
**Pull request by vinx13**
This PR added `lift_constants` option to FuseOpsByPattern since different BYOC backends have different requirements on whether constants should be lifted.

cc @masahi

### CODE REVIEW

File: tests/python/relax/test_codegen_cutlass.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def get_result_with_relax_cutlass_offload(mod, *args):
 
     seq = tvm.transform.Sequential(
         [
-            relax.transform.FuseOpsByPattern(patterns, annotate_codegen=True),
+            relax.transform.FuseOpsByPattern(patterns, lift_constants=True, annotate_codegen=True),
```
Comment (masahi) [inline tests/python/relax/test_codegen_cutlass.py:229]: I didn't realize that we have `partition_for_cutlass` for Relax, let's use that here.

**Review by masahi — COMMENTED** [review]
I prefer `bind_constants = True` to `lift_constants = False`. The name `lift_constants` refers to the inner working of `OperatorFusor` which shouldn't be exposed to users. From the user's perspective, what we want to say is "whether or not to keep bound constants in a grouped function". So I think `bind_constants` is more intuitive name (a more verbose option would be "keep_bound_constants"). And the default value for `bind_constants` should be True, since we shouldn't add unnecessary change to the input mod by default.

File: tests/python/relax/test_codegen_cutlass.py (hunk 1 above)
Comment (vinx13) [reply tests/python/relax/test_codegen_cutlass.py]: got some errors when using `partition_for_cutlass`, seems there are some issues with pattern registry, will fix and update it in a separate PR

File: python/tvm/relax/backend/contrib/cutlass.py [hunk 2]
```diff
@@ -117,4 +117,6 @@ def partition_for_cutlass(mod):
     """
 
     cutlass_patterns = get_patterns_with_prefix("cutlass")
-    return transform.FuseOpsByPattern(cutlass_patterns, annotate_codegen=True)(mod)
+    return transform.FuseOpsByPattern(cutlass_patterns, lift_constants=True, annotate_codegen=True)(
```
Comment (masahi) [inline python/tvm/relax/backend/contrib/cutlass.py:120]: needs update here

**Pull request merged**